application/controllers/admin/DataEntry.php

```diff
@@ -1325,6 +1329,23 @@ public function editdata($subaction, $id, $surveyid)
                         break;
                 }
 
+                if (!empty($questionInputs)) {
```

Comment:
This was added for short-text and single choice question here ? (i must test quickly)